dbms/src/Flash/EstablishCall.cpp

```diff
@@ -112,9 +112,19 @@ void EstablishCallData::writeErr(const mpp::MPPDataPacket & packet)
         err_status = grpc::Status(grpc::StatusCode::UNKNOWN, "Write error message failed for unknown reason.");
 }
 
-void EstablishCallData::writeDone(const ::grpc::Status & status)
+void EstablishCallData::setFinishState()
 {
     state = FINISH;
+    if (async_tunnel_sender && !async_tunnel_sender->isConsumerFinished())
+    {
+        async_tunnel_sender->consumerFinish(fmt::format("{}: setFinishState called.",
```

Comment:
Better to change `setFinishState()` to `setFinishState(const String & message)`, so we can distinguish different callers